internal/version/version.go

```diff
@@ -1,8 +1,41 @@
 package version
 
+import (
+	"fmt"
+	"runtime/debug"
+)
+
 // GitCommit indicates which commit the rukpak binaries were built from
-var GitCommit string
+var (
+	gitCommit  string = "unknown"
+	commitDate string = "unknown"
+	repoState  string = "unknown"
+)
 
 func String() string {
-	return GitCommit
+	return fmt.Sprintf("revision: %q, date: %q, state: %q", gitCommit, commitDate, repoState)
```

Comment:
Nit: Would it make sense to have a Go struct represent the version info and just print the object as a string? Not sure how that string would look off the top of my head and can always be done in a follow up if we think it would look nicer.

Reply:
Right now, we aren't exporting anything other than the string to print. Not sure how important that is, but given the only current use is to print a string, I figured I'd preserve that. If we need a struct and the ability to get the specific fields, we could reconsider when that need arises?

Reply:
I guess we exported `GitCommit`, but we didn't actually use it anywhere. Either way, adding the struct seemed to require more boilerplate and lines of code, so I kinda think what's here now is fine.